sdks/go/pkg/beam/io/webapi/webapi.go

```diff
@@ -0,0 +1,207 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webapi
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"reflect"
+)
+
+func init() {
+	beam.RegisterType(reflect.TypeOf((*wrappedCallerOnlyUserType)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*Request)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*Response)(nil)).Elem())
+	beam.RegisterDoFn(reflect.TypeOf((*callerFn)(nil)).Elem())
```

Comment:
Ideally use the generic registration package `register` instead. 

As a rule, don't use beam.RegisterDoFn anymore. It was a half measure before we had the generic solution.